### PR TITLE
[spark] support to read multi splits in a spark input partition

### PR DESCRIPTION
**Pull request by YannByron**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

**Comment by YannByron** [comment]
Without this pr, there are 115k+ tasks.
image

With this pr, it reduce to 10k+ tasks.
image

**Review by JingsongLi — APPROVED** [review]
+1 I have no comment. CC @Zouxxyy to take a look.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonInputPartition.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.spark
+
+import org.apache.paimon.table.source.Split
+
+import org.apache.spark.sql.connector.read.InputPartition
+
+case class PaimonInputPartition(splits: Array[Split]) extends InputPartition {}
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonInputPartition.scala:25]: we need to use `splits: Seq[Split]` to make sure case class equals

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonBatch.scala [hunk 2]
```diff
@@ -28,28 +28,28 @@ import java.util.Objects
...
-        this.splits.sameElements(other.splits) &&
+        this.inputPartitions.sameElements(other.inputPartitions) &&
         readBuilder.equals(other.readBuilder)
 
       case _ => false
     }
   }
 
   override def hashCode(): Int = {
-    Objects.hashCode(splits.toSeq, readBuilder)
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonBatch.scala:53]: make inputPartitions: `Array[PaimonInputPartition]` to Seq, then we can remove equals and hashcode

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/ScanHelper.scala [hunk 3]
```diff
@@ -120,14 +138,27 @@ trait ScanHelper extends Logging {
       .withPartition(split.partition())
       .withBucket(split.bucket())
       .withDataFiles(dataFiles.toList.asJava)
-      .rawConvertible(split.rawConvertible())
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/ScanHelper.scala:123]: revert it

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/sources/PaimonMicroBatchStream.scala [hunk 4]
```diff
@@ -122,7 +122,7 @@ class PaimonMicroBatchStream(
     val endOffset = PaimonSourceOffset(end)
 
     getBatch(startOffset, Some(endOffset), None)
-      .map(ids => new SparkInputPartition(ids.entry))
+      .map(ids => PaimonInputPartition(Array(ids.entry)))
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/sources/PaimonMicroBatchStream.scala:125]: -> `PaimonInputPartition(ids.entry)` for we alreadly have a apply method

**Review by Zouxxyy — APPROVED** [review]
+1

**Pull request merged**